src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -2922,14 +2922,20 @@ public struct StructEnum
         public void RefAssignment_Foreach_Nested()
         {
             verify(fieldType: "string?",
+                // (4,20): warning CS0649: Field 'C.Field' is never assigned to, and will always have its default value null
+                //     public string? Field;
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field").WithArguments("C.Field", "null").WithLocation(4, 20),
```

Comment:
>Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field").WithArguments("C.Field", "null").WithLocation(4, 20), [](start = 16, length = 112)

Why is this diagnostic now reported?

Reply:
Explanation in https://github.com/dotnet/roslyn/pull/36271#discussion_r292693391